src/R/Components/Impl/Application/Configuration/IO/ConfigurationSettingsWriter.cs

```diff
@@ -29,16 +30,18 @@ public sealed class ConfigurationSettingsWriter : IConfigurationSettingsWriter {
 
         /// <summary>
         /// Persists settings into R file. Settings are written
-        /// into R file as assignment statements such as 'name &lt;- value'.
+        /// into R file as assignment statements such as 'settings$name &lt;- value'.
         /// Value
         /// </summary>
         public void SaveSettings(IEnumerable<IConfigurationSetting> settings) {
             WriteHeader();
+            _writer.WriteLine("settings <- new.env()");
```

Comment:
`new.env()` creates an environment that has the current environment (probably `.GlobalEnv` here) as its parent; so if an identifier is not found, it will be looked in that parent. 

I would suggest rewriting this like so:
```r
settings <- as.environment(list(
    name1 = value1,
    ...,
    nameN = valueN));
```